src/ui/components/Library/Team/View/NewTestRuns/TestRunResultList.tsx

```diff
@@ -0,0 +1,50 @@
+import { useContext } from "react";
+
+import { TestRunTestWithRecordings } from "shared/test-suites/TestRun";
+
+import { useTestRunDetailsSuspends } from "../TestRuns/hooks/useTestRunDetailsSuspends";
+import { Alert } from "./Alert";
+import { TestResultListItem } from "./Overview/TestResultListItem";
+import { TestRunsContext } from "./TestRunsContextRoot";
+
+export function TestRunResultList({
+  selectedSpecTests,
+}: {
+  selectedSpecTests: TestRunTestWithRecordings[];
+}) {
+  const { testRunId } = useContext(TestRunsContext);
+  const { testRun } = useTestRunDetailsSuspends(testRunId);
+  const hasRecordings = selectedSpecTests.some(test =>
+    test.executions.some(e => e.recordings.length > 0)
+  );
```

Comment:
You can combine this and the map/flatMap operation below into a single, memoized loop to be more efficient.

1. Ignore any entries where `execution.recordings.length === 0`
2. Return the "no replays" message if the resulting flat-mapped collection is `length === 0`

Reply:
Tried this in https://github.com/replayio/devtools/pull/9986/commits/c66d724a8e2f990d94b18ce7bd73c2e97ab47281, but because of how that component accesses that data structure, it does mean that we have to compose the replay a bit weirdly. Let me know if this is what you had in mind!

Reply:
I don't think the commit you linked to is what I described.

I was suggested you could combine the `.some(condition)` and `.map(transform)` operations into a single `.forEach` operation, essentially

```js
const derived = [];
array.forEach(item =>
  if(condition) {
    derived.push(
      transform(item)
    );
   }
});
```

Reply:
Actually I just realized, if this isn't a blocking change do you mind if I merge it without?

We're moving up SCS-1591 and reopening #9954 to work on it next, which will undo a lot of this optimization anyway.